File: 547. Number_of_Provinces/solution.java
class Solution {
    
    public void DFS(int[][] isConnected, int cur_idx, boolean[] visited){
        
        for(int i = 0 ; i < isConnected.length ; i++){
            if(i != cur_idx && !visited[i] && isConnected[cur_idx][i] == 1){
                visited[i] = true;
                DFS(isConnected, i, visited);
            }
        }
        
        return;
        
    }
    
    
    public int findCircleNum(int[][] isConnected) {
        
        boolean[] visited = new boolean[isConnected.length];
        
        int ans = 0;
        
        for(int i = 0 ; i < isConnected.length ; i++){
            if(!visited[i]){
                visited[i] = true;
                DFS(isConnected, i, visited);
                ans++;
            }    
        }
        
        return ans;
        
    }
}

File: 547. Number_of_Provinces/solution.cpp
class Solution {
public:
    
    int ans = 0;
    
    void DFS(vector<vector<int>>& isConnected, int cur_idx, vector<int>& visited){
        
        for(int i = 0 ; i < isConnected[cur_idx].size() ; i++){
            if(isConnected[cur_idx][i] == 1 && !visited[i]){
                visited[i] = 1;
                DFS(isConnected, i, visited);
            }
                
        }
        
        return;
    }
    
    int findCircleNum(vector<vector<int>>& isConnected) {
       
        vector<int> visited(isConnected.size(), 0);
        
        for(int i = 0 ; i < isConnected.size() ; i++){
            if(!visited[i]){
                visited[i] = 1;
                DFS(isConnected, i, visited);
                ans++;
            }
        }
        
        
        return ans;
    }
};